src/packet_sources/macos.rs

```diff
@@ -0,0 +1,229 @@
+use crate::messages::{IpPacket, NetworkCommand, NetworkEvent, TunnelInfo};
+use crate::network::MAX_PACKET_SIZE;
+use crate::packet_sources::ipc::from_redirector::Message::Packet;
+use crate::packet_sources::ipc::{FromRedirector, PacketWithMeta};
+use crate::packet_sources::{ipc, PacketSourceConf, PacketSourceTask};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use home::home_dir;
+use nix::{sys::stat::Mode, unistd::mkfifo};
+use prost::Message;
+use std::io::Cursor;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tokio::net::unix::pipe;
+use tokio::sync::broadcast;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::mpsc::{unbounded_channel, Receiver, UnboundedReceiver, UnboundedSender};
+
+pub const IPC_BUF_SIZE: usize = MAX_PACKET_SIZE + 4;
+
+#[allow(dead_code)]
+pub struct PipeServer {
+    ip_rx: pipe::Receiver,
+    ip_tx: pipe::Sender,
+    net_rx: pipe::Receiver,
+    net_tx: pipe::Sender,
+    filter_rx: pipe::Receiver,
+    filter_tx: pipe::Sender,
+    ip_path: PathBuf,
+    net_path: PathBuf,
+    filter_path: PathBuf,
+}
```

Comment:
This may sound strange but we have to save both ends of the pipe somewhere because otherwise rust compiler notices that the variable is not used and it make it out of scope, and if it makes it out of scope it makes the pipe unreadable or unwritable and swift can't read or write from it